higra/attribute/tree_attributes.py

```diff
@@ -675,3 +675,43 @@ def attribute_topological_height(tree):
                                            hg.Accumulators.max)
 
     return res
+
+
+
+@hg.argument_helper(hg.CptHierarchy)
+@hg.auto_cache
+def attribute_moment_of_inertia(tree, altitudes, leaf_graph):
+    """
+    
+    Moment of inertia (first Hu moment) of each node of the given tree.   
+
+    :param tree: input tree (Concept :class:`~higra.CptHierarchy`)
+    :param altitudes: node altitudes of the input tree
+    :param leaf_graph: graph on the leaves of the input tree (deduced from :class:`~higra.CptHierarchy` on `tree`)
+    :return: a 1d array
+    """
+    coordinates = hg.attribute_vertex_coordinates(leaf_graph)
```

Comment:
There should be a test to check that ``leaf_graph`` is a 2d grid graph and throws a meaningful error if not. Something like:

```
if not hg.CptGridGraph.validate(leaf_graph) or len(hg.CptGridGraph.get_shape(leaf_graph)) != 2:
    raise ValueError(...)
```